src/core/response/parameters.rs

```diff
@@ -47,13 +57,80 @@ impl TryFrom<Json> for VpToken {
     type Error = Error;
 
     fn try_from(value: Json) -> Result<Self, Self::Error> {
-        serde_json::from_value(value).map(Self).map_err(Into::into)
+        match value {
+            // NOTE: When parsing a Json string object, it must be base64 encoded.
+            Json::String(s) => Ok(Self::Single(BASE64_STANDARD.encode(s.as_bytes()))),
+            // NOTE: When the Json is an object, it must be a map.
+            Json::Object(map) => Ok(Self::SingleAsMap(Json::Object(map))),
+            Json::Array(arr) => {
+                let mut tokens = Vec::new();
+                for value in arr {
+                    tokens.push(Self::try_from(value)?);
+                }
+                Ok(Self::Many(tokens))
+            }
+            _ => Err(Error::msg("Invalid vp_token")),
+        }
     }
 }
 
-impl From<VpToken> for Json {
-    fn from(value: VpToken) -> Self {
-        value.0.into()
+impl TryFrom<VpToken> for Json {
+    type Error = Error;
+
+    fn try_from(value: VpToken) -> Result<Self, Self::Error> {
+        match value {
+            VpToken::Single(s) => {
+                // Decode base64 string.
+                let bytes = BASE64_STANDARD.decode(s.as_bytes())?;
+                let s = String::from_utf8(bytes)?;
+
+                Ok(s.into())
+            }
+            VpToken::SingleAsMap(map) => Ok(map),
+            VpToken::Many(tokens) => {
+                let mut arr: Vec<Json> = Vec::new();
+                for token in tokens {
+                    arr.push(token.try_into()?);
+                }
+                Ok(arr.into())
+            }
+        }
+    }
+}
+
+impl TryFrom<VerifiablePresentation> for VpToken {
+    type Error = Error;
+
+    fn try_from(vp: VerifiablePresentation) -> Result<Self, Self::Error> {
+        Self::try_from(vp.0.into_serde_json())
+    }
+}
+
+impl TryFrom<VpToken> for Vec<VerifiablePresentation> {
+    type Error = Error;
```

Comment:
These are incorrect. `ssi::claims::jwt::VerifiablePresentation` cannot be converted into a VP token. It's the JWT claim for a `VerifiablePresentation` that you use if you are encoding a VP as a JWT. In reality the `vp_token` would be a JWT (that has this `VerifiablePresentation` claim), a [JsonPresentation](https://docs.rs/ssi-vc/latest/ssi_vc/syntax/enum.AnyJsonPresentation.html), or something else (for example ISO 18013-7 uses a CBOR encoded binary format for vp_tokens)

Reply:
Yeah, I wondered about using `JsonPresentation` type after reviewing Tim's SSI PR.

I'm going off of the https://openid.net/specs/openid-4-verifiable-presentations-1_0.html#section-6.1-2.2 definition. Is there a better definition to review. I didn't see anything about CBOR encoding.

Reply:
Commit [ae2c5c9](https://github.com/spruceid/oid4vp-rs/pull/18/commits/ae2c5c983fb15c4910a85c309498245a7d30bbfa) replaces the `VerifiablePresentation` with `AnyJsonPresentation`. It also removes the `VerifiablePresentatioBuilder` in favor of `JsonPresentation`